### DISCUSSION TITLE
How to remove vertical scrolling on Days content

### DISCUSSION THREAD

**Discussion by diegomotti** [ME-MarvinE/XCalendar#192] [Q&A]
Hi guys,

Thank you for developing xCalendar—it's been really helpful in my project! I have a question regarding gesture handling. Is there a way to implement horizontal or vertical gestures within the xCalendar component? For example, I'd like to allow users to swipe left/right to navigate between months or up/down to toggle between month view and week view.

If possible, could you guide me on how to achieve this?

Thanks in advance for your help!

**Comment by ME-MarvinE** [comment]
I would say to use the `CalendarView`'s `DaysViewTemplate` property. Put a `ContentPresenter` in it or a `DaysView` with the appropriate properties (`Days`, `DaysOfWeek`, and `DayTemplate`) and assign them a `SwipeGestureRecognizer` executing whatever command you use to perform navigation of the calendar.

However, there are issues with the `CollectionView` that prevent GestureRecognizers from working correctly. They've been present since 2022 so I don't think GestureRecognizers will be an option anytime soon:
dotnet/maui#10213
dotnet/maui#25758

That being said, there is an example in the sample app called "Animated Swipable Calendar" which does exactly what you're asking: navigating between months by swiping horizontally (You can modify the code to do it vertically instead if needed).

It uses the work around of having multiple calendars defined, each being rendered as items inside a `CollectionView` which then provides the scrolling behaviour.

While trying to get GestureRecognizers to work, I did also find an issue in the `CalendarView` and `DaysView` which prevents the `DaysViewTemplate` from detecting changes to the `Days` property due to how it's updated internally. More info in this issue: #193.
